packages/api-explorer/src/scenes/DiffScene/DiffScene.spec.tsx

```diff
@@ -0,0 +1,162 @@
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ */
+import React from 'react'
+import { screen, waitFor, render } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+
+import type { SpecItem, SpecList } from '@looker/sdk-codegen'
+import { useLocation } from 'react-router-dom'
+import { useDispatch } from 'react-redux'
+import { getLoadedSpecs } from '../../test-data'
+import {
+  createTestStore,
+  renderWithRouter,
+  renderWithRouterAndReduxProvider,
+} from '../../test-utils'
+import { getApixAdaptor, MockedProvider, mockHistory } from '../../utils'
+import { DiffScene } from './DiffScene'
+
+// jest.mock('react-redux', () => {
+//   const reactRedux = jest.requireActual('react-redux')
+//   return {
+//     __esModule: true,
+//     ...reactRedux,
+//     useDispatch: jest.fn(reactRedux.useDispatch),
+//   }
+// })
+
+// jest.mock('react-router-dom', () => {
+//   const ReactRouter = jest.requireActual('react-router-dom')
+//
+//   return {
+//     __esModule: true,
+//     ...ReactRouter,
+//     useHistory: jest.fn().mockReturnValue({
+//       push: jest.fn(),
+//       location,
+//       useLocation: jest
+//         .fn()
+//         .mockReturnValue({ pathname: '/4.0/diff/3.1', search: '' }),
+//     }),
+//   }
+// })
+
+const specs = getLoadedSpecs() as SpecList
+// class MockApixAdaptor {
+//   async fetchSpec(spec: SpecItem) {
+//     return new Promise(() => specs[spec.key])
+//   }
+// }
+
+// const mockApixAdaptor = new MockApixAdaptor()
+// jest.mock('../../utils/apixAdaptor', () => {
+//   const apixAdaptor = jest.requireActual('../../utils/apixAdaptor')
+//   return {
+//     __esModule: true,
+//     ...apixAdaptor,
+//     getApixAdaptor: jest.fn(),
+//   }
+// })
+
+describe('DiffScene', () => {
+  // const mockDispatch = jest.fn()
+
+  // afterEach(() => {
+  //   jest.clearAllMocks()
+  // })
+  // ;(getApixAdaptor as jest.Mock).mockReturnValue(mockApixAdaptor)
+  Element.prototype.scrollTo = jest.fn()
+  Element.prototype.scrollIntoView = jest.fn()
+
+  const toggleNavigation = () => false
+  test.only('updating url dispatches store action', () => {
+    // ;(useDispatch as jest.Mock).mockReturnValue(mockDispatch)
+    // TODO issue: executing createTestStore here doesn't init default settings
+    const store = createTestStore({
```

Comment:
Issue: Having a test that just executes `createTestStore` in `DiffScene.spec.tsx` will fail to initialize the store with default settings, only taking values from the overrides.